senior-knights/course-schedulizer
# save app state between sessions

This saves the app state between sessions in localStorage. This means that it will only save the app state for the current web browser. This allows the user to leave the web page and come back and continue editing where they left off. 

To test:
- `npm start`
- Upload a schedule
- Refresh the page or navigate backward and notice the schedule loads when you return
- Change the semester and refresh again; notice the state persists through app sessions

This PR:
- a `useLocal` hook to help with modifying values in local storage
-   helpers for the hook
- updates to `initalAppState`
- changes to the main `reducer` function to be a closure that will accept a callback function when initialized

closes #99 

Please let me know any comments or changes!
